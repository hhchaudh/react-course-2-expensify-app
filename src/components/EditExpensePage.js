import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
  onSubmit = (expenseUpdate) => {
    this.props.startEditExpense(this.props.expense.id, expenseUpdate);
    this.props.history.push('/');
  };

  onRemove = () => {
    this.props.startRemoveExpense({id: this.props.expense.id});
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
        <ExpenseForm
          expense={this.props.expense}
          onSubmit={this.onSubmit}
        />
        <button onClick={this.onRemove}>Remove</button>
      </div>
    );
  }
}

// const EditExpensePage = (props) => {
//   return (
//     <div>
//       <ExpenseForm
//         expense={props.expense}
//         onSubmit={(expenseUpdate) => {
//           props.dispatch(editExpense(props.expense.id, expenseUpdate));
//           props.history.push('/');
//         }}
//       />
//       <button onClick={() => {
//         props.dispatch(removeExpense({ id: props.expense.id }));
//         props.history.push('/');
//       }}>Remove</button>
//     </div>
//   );
// };

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  };
};

const mapDispatchToProps = (dispatch) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
