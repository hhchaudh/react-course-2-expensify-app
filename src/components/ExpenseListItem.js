import React from 'react';
import {Link} from 'react-router-dom';


const ExpenseListItem = ({id, description, amount, createdAt}) => (
  <div>
    <Link to={`/edit/${id}`}><h3>{description}</h3></Link>
    <p>{amount} - {createdAt}</p>
  </div>
);


export default ExpenseListItem;


// export default ExpenseListItem;

// const mapStateToProps = (state) => {
//   return {
//     expenses: state.expenses
//   };
// };
//
// export default connect(mapStateToProps)(ExpenseListItem);


//   { props.expenses.map((expense) => (
//     expense.description
// ))}
// { props.expenses.map((expense) => (
//   expense.amount
// ))}
