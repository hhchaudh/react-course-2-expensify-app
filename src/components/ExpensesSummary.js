import * as React from 'react';
import pluralize from 'pluralize';
import expenseSum from '../selectors/expenses-total';
import selectExpense from '../selectors/expenses';
import numeral from 'numeral';
import { connect } from 'react-redux';

export const ExpenseSummary = (props) => (
  <div>
    {
      props.expenses.length === 0 ?
      <p>No Expenses to view</p> :
      <p>{`Viewing ${props.expenses.length} ${pluralize('expense', props.expenses.length)} totalling ${numeral(expenseSum(props.expenses) / 100).format('$0,0.00')}`}</p>
    }
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpense(state.expenses, state.filters)
  }
};

export default connect(mapStateToProps)(ExpenseSummary);
