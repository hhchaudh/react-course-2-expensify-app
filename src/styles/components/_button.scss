@mixin button {
  background-color: $blue;
  border: none;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}

.button {
  @include button;
}

.button--remove {
  @include button;
  background-color: #888888;
}
