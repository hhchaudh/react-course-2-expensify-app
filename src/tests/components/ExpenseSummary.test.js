import * as React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from "../../components/ExpensesSummary";
import expenses from '../fixtures/expenses';

test('should render ExpenseSummary header with no expenses correctly', () => {
  let wrapper = shallow(<ExpenseSummary expenses={[]}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary header with no expenses correctly', () => {
  let wrapper = shallow(<ExpenseSummary expenses={expenses}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should pluralize correctly', () => {
  let wrapper = shallow(<ExpenseSummary expenses={[expenses[0]]}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should have correct total and output for all expenses', () => {
  let wrapper = shallow(<ExpenseSummary expenses={expenses}/>);
  expect(wrapper.find('h1').text()).toEqual('Viewing 3 expenses totalling $1,141.95');
});

test('should have correct total and output for one expense', () => {
  let wrapper = shallow(<ExpenseSummary expenses={[expenses[0]]}/>);
  expect(wrapper.find('h1').text()).toEqual('Viewing 1 expense totalling $1.95');
});

test('should have correct total for all expenses', () => {
  let wrapper = shallow(<ExpenseSummary expenses={[]}/>);
  expect(wrapper.find('h1').text()).toEqual('Viewing 0 expenses totalling $0.00');
});
