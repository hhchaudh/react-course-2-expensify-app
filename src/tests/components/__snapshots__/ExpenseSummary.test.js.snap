// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should pluralize correctly 1`] = `
<div>
  <p>
    Viewing 1 expense totalling $1.95
  </p>
</div>
`;

exports[`should render ExpenseSummary header with no expenses correctly 1`] = `
<div>
  <p>
    No Expenses to view
  </p>
</div>
`;

exports[`should render ExpenseSummary header with no expenses correctly 2`] = `
<div>
  <p>
    Viewing 3 expenses totalling $1,141.95
  </p>
</div>
`;
